Dikke Ploaten/Controller/SettingsTableViewController.swift

```diff
@@ -29,4 +54,94 @@ class SettingsTableViewController : UITableViewController {
 		let initial = storyboard.instantiateInitialViewController()
 		UIApplication.shared.keyWindow?.rootViewController = initial
 	}
+	
+	// MARK: - UIImagePickerControllerDelegate Methods
+	
+	func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+		if let image = info[UIImagePickerController.InfoKey.originalImage] as? UIImage {
+			if(isImgProfile) {
+				imgProfile.contentMode = .scaleAspectFit
+				imgProfile.image = image
+			} else {
+				imgBackgroundCover.contentMode = .scaleAspectFit
+				imgBackgroundCover.image = image
+			}
+		}
+		if(isImgProfile) {
+			Database.shared.uploadImage(image: imgProfile.image!, isImgProfile: true)
+			isImgProfile = false
+		} else {
+			Database.shared.uploadImage(image: imgBackgroundCover.image!, isImgProfile: false)
+		}
+		
+		imagePicker.dismiss(animated: true, completion: nil)
+	}
+	
+	func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+		dismiss(animated: true, completion: nil)
+	}
+	
+	// MARK: - TableView
+	
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		tableView.deselectRow(at: indexPath, animated: false)
+		if indexPath.section == 0 {
+			if indexPath.row == 0 {
+				isImgProfile = true
+			}
+			imagePicker.allowsEditing = false
+			imagePicker.sourceType = .photoLibrary
+			
+			present(imagePicker, animated: true, completion: nil)
+			
+		}
+		if indexPath.section == 1 {
+			if indexPath.row == 0 {
+				let alertController = UIAlertController(title: "Verander naam", message: "Verander hier je gebruikersnaam:", preferredStyle: .alert)
+				alertController.addTextField { textField in
+					textField.placeholder = "Gebruikersnaam"
+				}
+				let confirmAction = UIAlertAction(title: "OK", style: .default) { [weak alertController] _ in
+					guard let alertController = alertController, let textField = alertController.textFields?.first else { return }
+					Database.shared.updateUsername(username: textField.text!, completionHandler: { err in
+						if let err = err {
+							print(err)
+						}
+						Database.shared.getUser { user in
+							self.lblName.text = user.username
+							self.lblEmail.text = user.email
+						}
+					})
+				}
+				alertController.addAction(confirmAction)
+				let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+				alertController.addAction(cancelAction)
+				present(alertController, animated: true, completion: nil)
+			}
+			if indexPath.row == 1{
+				let alertController = UIAlertController(title: "Verander wachtwoord", message: "Verander hier je wachtwoord:", preferredStyle: .alert)
+				alertController.addTextField { textField in
```

Comment:
Waar is de "oud wachtwoord", "nieuw wachtwoord", "herhaal wachtwoord"?

Reply:
Nja ik ben niet zo voor wachtwoorden opslaan, en ziet er niet uit alsof Firestore ondersteunt om nieuw wachtwoord te kiezen met verificatie van het oude. Maak dus maar 2 tekstveldjes "nieuw" en "herhaal"